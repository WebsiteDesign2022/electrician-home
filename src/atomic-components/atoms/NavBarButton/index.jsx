import React from 'react';
import { Nav } from 'react-bootstrap';

const NavBarButton = ({ link, displayName }) => (
    <Nav.Link className="" href={link}>{displayName}</Nav.Link>
);

export default NavBarButton;
