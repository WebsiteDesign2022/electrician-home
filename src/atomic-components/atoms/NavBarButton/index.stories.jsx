import React from 'react';
import 'bootstrap/dist/css/bootstrap.min.css';

import NavBarButton from './index';

export default {
    title: 'Electrician-Home/atoms/NavBarButton',
    component: NavBarButton,
    parameters: {
        layout: 'fullscreen'
    }
};

const Template = (args) => <NavBarButton {...args} />;

// More on args: https://storybook.js.org/docs/react/writing-stories/args

export const Primary = Template.bind({});
Primary.args = {
    link: 'link', 
    displayName: 'display name' 
};
