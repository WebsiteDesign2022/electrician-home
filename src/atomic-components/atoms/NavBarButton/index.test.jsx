import {render, screen} from '@testing-library/react'
import TestRenderer from 'react-test-renderer';

import NavBarButton from '.';

describe("NavBarButton", () => {
    test("renders NavBarButton and contains link", () => {
    
        render(<NavBarButton link="www.google.com" displayName="Google" />);

        expect(
            screen.getByRole("link", { link: 'www.google.com', displayName: 'Google' })
        ).toBeVisible();

        expect(
            screen.getByText(/Google/i)
        ).toBeInTheDocument();

    });

    test("renders NavBarButton", () => {
        const navBarButton = TestRenderer.create(
            <NavBarButton/>
        );

        expect(navBarButton.toJSON()).toMatchSnapshot();
    });
});
