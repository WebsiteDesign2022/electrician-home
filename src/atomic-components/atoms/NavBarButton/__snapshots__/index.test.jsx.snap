// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBarButton renders NavBarButton 1`] = `
<a
  className="nav-link"
  href="#"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex={0}
/>
`;
